app/src/main/java/protect/card_locker/ManageGroupActivity.java

```diff
@@ -0,0 +1,315 @@
+            } else {
+                mGroupNameNotInUse = true;
+            }
+        }
+    }
+
+    @Override
+    protected void onResume()
+    {
+        super.onResume();
+
+        setTitle(getString(R.string.edit) + ": " + mGroup._id);
+
+        if (mCurrentGroupName == null){
+            mGroupNameText.setText(mGroup._id);
+        }else{
+            mGroupNameText.setText(mCurrentGroupName);
+            checkIfGroupNameIsInUse();
+        }
+
+        mGroupNameText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                return;
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                return;
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                resetGroupNameTextColor();
+                checkIfGroupNameIsInUse();
+            }
+        });
+
+        if (mAdapterState != null){
+            mAdapter.importInGroupState(mAdapterState);
+        }
+        updateLoyaltyCardList();
+    }
+
+    @Override
+    protected void onSaveInstanceState (Bundle outState){
+        super.onSaveInstanceState(outState);
+        if(mAdapterState != null){
+            ManageGroupActivityInGroupState state = new ManageGroupActivityInGroupState(mAdapterState);
+            outState.putParcelable("mAdapterState", state);
+        }
+        if(mCurrentGroupName != null){
+            outState.putString("mCurrentGroupName", mCurrentGroupName);
+        }
+
+    }
+
+    @Override
+    protected void onPause(){
+        super.onPause();
+
+        mAdapterState = mAdapter.exportInGroupState();
+        mCurrentGroupName = mGroupNameText.getText().toString();
+    }
+
+    private void updateLoyaltyCardList() {
+        mAdapter.swapCursor(mDB.getLoyaltyCardCursor(mFilter, null, mOrder, mOrderDirection));
+
+        if(mAdapter.getCountFromCursor() > 0)
+        {
+            // We want the cardList to be visible regardless of the filtered match count
+            // to ensure that the noMatchingCardsText doesn't end up being shown below
+            // the keyboard
+            mHelpText.setVisibility(View.GONE);
+            mNoGroupCardsText.setVisibility(View.GONE);
+            if(mAdapter.getItemCount() > 0)
+            {
+                mCardList.setVisibility(View.VISIBLE);
+                mNoMatchingCardsText.setVisibility(View.GONE);
+            }
+            else
+            {
+                mCardList.setVisibility(View.GONE);
+                if (!mFilter.isEmpty()) {
+                    // Actual Empty Search Result
+                    mNoMatchingCardsText.setVisibility(View.VISIBLE);
+                    mNoGroupCardsText.setVisibility(View.GONE);
+                } else {
+                    // Group Tab with no Group Cards
```

Comment:
// Group Tab with no Group Cards

Should never be a thing that happens

Also, right now, if there are no cards editing a group shows incorrect instructions:
![image](https://user-images.githubusercontent.com/1885159/139580609-86c3ca51-5cbe-4eeb-8802-2fd8a6230520.png)

Should probably has its own text like "After creating some cards, you can edit which cards are part of this group here"